agent/consul/intention_endpoint.go

```diff
@@ -783,48 +783,6 @@ func (s *Intention) Check(
 		}
 	}
 
-	// Get the matches for this destination
```

Comment:
You might want to stare at the agent authorize endpoint (agent/agent_endpoint.go) to see if you can use your new state store function there to cut down on duplication that already exists.